File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
         int [] arr = new int[2];
        HashMap<Integer,Integer>hm = new HashMap<>();
        for (int i=0;i<nums.length;i++){
            int x=target-nums[i];
            if (hm.containsKey(x)){
               
                arr[0]=i;
                arr[1]=hm.get(x);
                return arr;
            }
            hm.put(nums[i],i);
        }
        return arr;
    }
}

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>mp;
        vector<int>v;
        int n=nums.size();
        for (int i=0;i<n;i++){
            if (mp.find(target-nums[i])!=mp.end()){
                v.push_back(i);
                v.push_back(mp[target-nums[i]]);
            }
            mp[nums[i]]=i;
        }
        return v;
    }
};